client/src/components/tags/TagTree/TagTree.tsx

```diff
@@ -42,9 +43,11 @@ export default function TagTree({
 						Soon, you will be able to edit the hierarchy of your tags from here.
 					</p>
 					<S.Tree $orientation={orientation} $columnCount={rootTags.length}>
-						{rootTags.map((tag) => (
-							<Tag key={tag.tag_id} tag={tag} level={0} />
-						))}
+						{rootTags
+							.filter((t) => !!t)
```

Comment:
Do the filtering when defining rootTags. Same comment for the other case where this applies.